### ISSUE TITLE
<DataGrid>  and <NoDataEmptyState>  Accessibility test failure

### ISSUE THREAD

**Issue by SeonyuK** [carbon-design-system/ibm-products#3731]
## What package(s) are you using?

- [X] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## Detailed description

> Describe in detail the issue you're having.

- Datagrid and NoDataEmptyState are failing the accessibility tests. 

- In the **Datagrid**

Rule violation: #aria-allowed-role [minor]
    > Ensures role attribute has an appropriate value for the element
    ARIA role should be appropriate for the element
    https://dequeuniversity.com/rules/axe/4.4/aria-allowed-role?application=axeAPI
    ================================================================================
    Node:
    <input draggable="false" role="separator" style="cursor: col-resize;" class="c4p--datagrid__col-resizer-range" type="range" aria-label="Resize column" value="240">
    Fix any of the following:
      ARIA role separator is not allowed for given element

- In the **NoDataEmptyState**

Rule violation: #svg-img-alt [serious]
    > Ensures <svg> elements with an img, graphics-document or graphics-symbol role have an accessible text
    <svg> elements with an img role must have an alternative text
    https://dequeuniversity.com/rules/axe/4.4/svg-img-alt?application=axeAPI
    ================================================================================
    Node:
    <svg xmlns="http://www.w3.org/2000/svg" width="80" height="80" viewBox="0 0 80 80" class="c4p--empty-state__illustration c4p--empty-state__illustration-noData c4p--empty-state__illustration--lg" role="img">
    Fix any of the following:
      Element has no child that is a title
      aria-label attribute does not exist or is empty
      aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
      Element has no title attribute

> Is this issue related to a specific component?

- Datagrid

> What did you expect to happen? What happened instead? What would you like to
> see changed?

- No fail for the accessibility test

> What browser are you working in?

> What version of the @carbon/ibm-products (or @carbon/ibm-cloud-cognitive)
> package are you using?

- We are using @carbon/ibm-products": "2.13.0

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

- CP4S.

## Steps to reproduce the issue

1. Step one
2. Step two
3. Step three
4. etc.

> Please create a reduced test case in CodeSandbox
>
> https://codesandbox.io/s/github/carbon-design-system/ibm-products/tree/main/examples/carbon-for-ibm-products/SidePanel

## Additional information

- Screenshots or code
- Notes

**Comment by SeonyuK** [comment]
I found more failures for the Datagrid component 

==========================================

- Message: Radio input is in a different group than another Radio with the name "row-height-group"
      Level: violation
      XPath: /html[2]/body[1]/main[1]/div[1]/div[1]/div[1]/section[1]/div[1]/span[1]/span[2]/span[1]/div[1]/div[1]/div[1]/fieldset[1]/div[5]/input[1]
      Snippet: <input value="xl" name="row-height-group" id="datagrid--row-density--xl" class="cds--radio-button" type="radio">
      Help: https://able.ibm.com/rules/archives/latest/doc/en-US/input_checkboxes_grouped.html#%7B%22message%22%3A%22Radio%20input%20is%20in%20a%20different%20group%20than%20another%20Radio%20with%20the%20name%20%5C%22row-height-group%5C%22%22%2C%22snippet%22%3A%22%3Cinput%20value%3D%5C%22xl%5C%22%20name%3D%5C%22row-height-group%5C%22%20id%3D%5C%22datagrid--row-density--xl%5C%22%20class%3D%5C%22cds--radio-button%5C%22%20type%3D%5C%22radio%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_NotSameGroup%22%2C%22ruleId%22%3A%22input_checkboxes_grouped%22%7D
    - Message: Radio input is in a different group than another Radio with the name "row-height-group"
      Level: violation
      XPath: /html[2]/body[1]/main[1]/div[1]/div[1]/div[1]/section[1]/div[1]/span[1]/span[2]/span[1]/div[1]/div[1]/div[1]/fieldset[1]/div[4]/input[1]
      Snippet: <input checked="" value="lg" name="row-height-group" id="datagrid--row-density--lg" class="cds--radio-button" type="radio">
      Help: https://able.ibm.com/rules/archives/latest/doc/en-US/input_checkboxes_grouped.html#%7B%22message%22%3A%22Radio%20input%20is%20in%20a%20different%20group%20than%20another%20Radio%20with%20the%20name%20%5C%22row-height-group%5C%22%22%2C%22snippet%22%3A%22%3Cinput%20checked%3D%5C%22%5C%22%20value%3D%5C%22lg%5C%22%20name%3D%5C%22row-height-group%5C%22%20id%3D%5C%22datagrid--row-density--lg%5C%22%20class%3D%5C%22cds--radio-button%5C%22%20type%3D%5C%22radio%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_NotSameGroup%22%2C%22ruleId%22%3A%22input_checkboxes_grouped%22%7D
    - Message: Radio input is in a different group than another Radio with the name "row-height-group"
      Level: violation
      XPath: /html[2]/body[1]/main[1]/div[1]/div[1]/div[1]/section[1]/div[1]/span[1]/span[2]/span[1]/div[1]/div[1]/div[1]/fieldset[1]/div[3]/input[1]
      Snippet: <input value="md" name="row-height-group" id="datagrid--row-density--md" class="cds--radio-button" type="radio">
      Help: https://able.ibm.com/rules/archives/latest/doc/en-US/input_checkboxes_grouped.html#%7B%22message%22%3A%22Radio%20input%20is%20in%20a%20different%20group%20than%20another%20Radio%20with%20the%20name%20%5C%22row-height-group%5C%22%22%2C%22snippet%22%3A%22%3Cinput%20value%3D%5C%22md%5C%22%20name%3D%5C%22row-height-group%5C%22%20id%3D%5C%22datagrid--row-density--md%5C%22%20class%3D%5C%22cds--radio-button%5C%22%20type%3D%5C%22radio%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_NotSameGroup%22%2C%22ruleId%22%3A%22input_checkboxes_grouped%22%7D
    - Message: Radio input is in a different group than another Radio with the name "row-height-group"
      Level: violation
      XPath: /html[2]/body[1]/main[1]/div[1]/div[1]/div[1]/section[1]/div[1]/span[1]/span[2]/span[1]/div[1]/div[1]/div[1]/fieldset[1]/div[2]/input[1]
      Snippet: <input value="sm" name="row-height-group" id="datagrid--row-density--sm" class="cds--radio-button" type="radio">
      Help: https://able.ibm.com/rules/archives/latest/doc/en-US/input_checkboxes_grouped.html#%7B%22message%22%3A%22Radio%20input%20is%20in%20a%20different%20group%20than%20another%20Radio%20with%20the%20name%20%5C%22row-height-group%5C%22%22%2C%22snippet%22%3A%22%3Cinput%20value%3D%5C%22sm%5C%22%20name%3D%5C%22row-height-group%5C%22%20id%3D%5C%22datagrid--row-density--sm%5C%22%20class%3D%5C%22cds--radio-button%5C%22%20type%3D%5C%22radio%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_NotSameGroup%22%2C%22ruleId%22%3A%22input_checkboxes_grouped%22%7D
    - Message: Radio input is in a different group than another Radio with the name "row-height-group"
      Level: violation
      XPath: /html[2]/body[1]/main[1]/div[1]/div[1]/div[1]/section[1]/div[1]/span[1]/span[2]/span[1]/div[1]/div[1]/div[1]/fieldset[1]/div[1]/input[1]
      Snippet: <input value="xs" name="row-height-group" id="datagrid--row-density--xs" class="cds--radio-button" type="radio">
      Help: https://able.ibm.com/rules/archives/latest/doc/en-US/input_checkboxes_grouped.html#%7B%22message%22%3A%22Radio%20input%20is%20in%20a%20different%20group%20than%20another%20Radio%20with%20the%20name%20%5C%22row-height-group%5C%22%22%2C%22snippet%22%3A%22%3Cinput%20value%3D%5C%22xs%5C%22%20name%3D%5C%22row-height-group%5C%22%20id%3D%5C%22datagrid--row-density--xs%5C%22%20class%3D%5C%22cds--radio-button%5C%22%20type%3D%5C%22radio%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_NotSameGroup%22%2C%22ruleId%22%3A%22input_checkboxes_grouped%22%7D

**Comment by SeonyuK** [comment]
Another one

 - Message: An explicit ARIA 'role' is not valid for <tr> element within a ARIA role 'table' per the ARIA in HTML specification
      Level: violation
      XPath: /html[2]/body[1]/main[1]/div[1]/div[1]/div[2]/div[1]/div[1]/table[1]/tbody[1]/tr[1]
      Snippet: <tr style="display: flex; flex: 1 0 auto; min-width: 0px;" role="row" class="c4p--datagrid__carbon-row">
      Help: https://able.ibm.com/rules/archives/latest/doc/en-US/table_aria_descendants.html#%7B%22message%22%3A%22An%20explicit%20ARIA%20'role'%20is%20not%20valid%20for%20%3Ctr%3E%20element%20within%20a%20ARIA%20role%20'table'%20per%20the%20ARIA%20in%20HTML%20specification%22%2C%22snippet%22%3A%22%3Ctr%20style%3D%5C%22display%3A%20flex%3B%20flex%3A%201%200%20auto%3B%20min-width%3A%200px%3B%5C%22%20role%3D%5C%22row%5C%22%20class%3D%5C%22c4p--datagrid__carbon-row%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22explicit_role%22%2C%22ruleId%22%3A%22table_aria_descendants%22%7D

**Comment by davidmenendez** [comment]
some of the error's you're seeing are specifically storybook related. the rest are valid and luckily i've taken care of them in the V2 refactor of empty state. there's an [open PR](https://github.com/carbon-design-system/ibm-products/pull/3741) which addresses existing accessibility concerns.

hopefully we can get it merged soon.

**Comment by SeonyuK** [comment]
Those errors are from our product aat tests not from the storybook. 
As soon as it gets delivered, we will update the version and will update you on what happens

**Comment by davidmenendez** [comment]
just needs a V1 fix and will close

**Comment by davidmenendez** [comment]
@SeonyuK fixes should show up in the next release

**Comment by SeonyuK** [comment]
Thanks! I will update the version today!!!!

**Comment by SeonyuK** [comment]
@davidmenendez  Are these fixes included in v2.16.0?
We updated the version to 2.16.0 and still can see the issues

**Comment by davidmenendez** [comment]
i have verified that the specific issues regarding empty state and datagrid have been fixed. the accessibility scanner still shows issues that need to be addressed, but they aren't empty state specific. they are more general issues related to datagrid. @elycheea not sure if we should close this issue to track those additional accessibility violations for datagrid or close this as only pertaining to empty state.

**Comment by elycheea** [comment]
The other accessibility violations are documented in #3874. @SeonyuK can you confirm that the issues is also fixed for you before we close this? We can use the other issue to track the follow up fixes.

**Comment by elycheea** [comment]
@all-contributors please add @SeonyuK for accessibility 🙌

**Comment by SeonyuK** [comment]
I will when it gets released next week and will let you know

**Comment by davidmenendez** [comment]
thanks @SeonyuK

**Comment by SeonyuK** [comment]
After updating to v2.17.1, I can see the <title> in the SVG is filled. However, our AAT test keeps failing so I dug more and found it is actually a jest bug.
Here, you can check here https://github.com/jsdom/jsdom/issues/3064
So there is nothing we can do anymore. I am good to close it if you want
